calculator_test.go

```diff
@@ -2,23 +2,275 @@ package calculator_test
 
 import (
 	"calculator"
+	"math/rand"
 	"testing"
 )
 
-func TestAdd(t *testing.T) {
-	t.Parallel()
-	var want float64 = 4
-	got := calculator.Add(2, 2)
+func assertNumbers(want, got float64, t *testing.T) {
 	if want != got {
-		t.Errorf("want %f, got %f", want, got)
+		t.Fatalf("want %f, got %f", want, got)
+	}
+}
+
+func assertNumbersErrHandling(want, got float64, errExpected bool, t *testing.T) {
+	if (want != got) && !errExpected {
```

Comment:
There's a problem here, though (and it would be easier to see if this logic were right in the test). We know `err` is `nil`, or the test would already have failed. But what if the test case is _supposed_ to return an error and it doesn't? For example, the divide by zero case. We don't catch that case.

Reply:
Not sure what you mean, If I remove errExpected it catches the error:
```
Running tool: /usr/local/bin/go test -timeout 30s calculator -run ^TestDivide$

--- FAIL: TestDivide (0.00s)
    --- FAIL: TestDivide/Divide_by_zero (0.00s)
        calculator_test.go:206: Cannot divide inputs 6.00 0.00 []: Cannot divide by zero
FAIL
FAIL	calculator	0.720s
FAIL
```

Reply:
You'll see what I mean if you comment out these lines in Divide:

```go
	// if n2 == 0 {
	// 	return 0, errors.New(divideByZeroError)
	// }
```

This stops Divide returning error when its second argument is zero. Now run the tests again. They should fail, shouldn't they? But they don't:

```
go test
PASS
ok      calculator      0.183s
```